synapse/app/media_repository.py

```diff
@@ -71,13 +73,20 @@ def _listen_http(self, listener_config):
                     resources[METRICS_PREFIX] = MetricsResource(RegistryProxy)
                 elif name == "media":
                     media_repo = self.get_media_repository_resource()
+
+                    # We need to serve the admin servlets for media on the
+                    # worker.
+                    admin_resource = JsonResource(self, canonical_json=False)
+                    register_servlets_for_media_repo(self, admin_resource)
+
                     resources.update(
                         {
                             MEDIA_PREFIX: media_repo,
                             LEGACY_MEDIA_PREFIX: media_repo,
                             CONTENT_REPO_PREFIX: ContentRepoResource(
                                 self, self.config.uploads_path
                             ),
+                            "/_synapse/admin": admin_resource,
```

Comment:
this seems to be sprawling a bit beyond the PR title and changelog comment. Please could the admin stuff be kept separate?

Reply:
ohhh I see, it's the admin APIs that pull in the `MediaRepository` which runs the background jobs. sorry, ignore this, I'll have another look.